### DISCUSSION TITLE
[Feature] Manually create categories on the people page

### DISCUSSION THREAD

**Discussion by sujaldev** [immich-app/immich#6079] [Feature Request]
### The feature

Could we have a feature to manually categorize people into groups like say family, friends, colleagues, etc. I'd assume this feature being implemented would also help in #6078.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by aisbergde** [comment]
a possible way: tags associated with persons / faces, and hierarchical tags to group them. Then this could be solved with #2168

**Comment by imler** [comment]
I would love to see that because it would dramatically declutter the "People" page and allow for easier finding of specific people.

My two cents about implementation though:

1. Would be great if each category could have subcategories, so it would be possible to further split one group - i.e. family into smaller more specific groups

2. It should be possible to assign one person to multiple groups. Imagine a situation where someone from a family is also working in the same company

**Reply by CrushedAsian255** [reply]
Yes I think this is a good idea.

**Reply by jarrodmoldrich** [reply]
Being able to assign tags to people would solve point 2.  Then, hierarchical tags as mentioned in the [other comment](https://github.com/immich-app/immich/discussions/6079#discussioncomment-7984424) would solve point 1

**Comment by sgofferj** [comment]
I'd say, integrating it with tags would be best from an UX perspective. The search function could for a tag search then return first people and after that assets with the corresponding tag.

**Comment by roshavi4ak** [comment]
I would love this functionallity also

**Comment by Hankaba** [comment]
I would also like to see a tag or some other way to mark people who are shown as a person but are not actually a person, or who are irrelevant to the library as a face in a crowd. People tagged with this tag should be able to be permanently removed from all views. This would make working with people a lot easier.

**Comment by dionorgua** [comment]
Digikam associates tags for faces. All faces are by default child tags of "People" root tag. But it's possible to make any hierarchy. Plus digikam shows avatar as tag icon.

And this is good thing because it firstly allows search for for all "Family" members, or ex-coworkers, etc. Plus it's possible to just 'tag' photo with `People/Name` tag for cases where it's impossible to select actual face on photo (like covered or not visible at all) but when it still makes sense to show that photo in search results.

Ideally Immich should do same

**Comment by jarrodmoldrich** [comment]
@dionorgua Absolutely this!  The Immich schema could really take inspiration from `git`.  Git is basically just an [object store](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects) with [references](https://git-scm.com/book/en/v2/Git-Internals-Git-References).  Everything is either a blob (an image), or something that points to blobs or other pointers to blobs (a tree/folder), or something that points to a pointer of a blob, which for Immich could be an album, or a person, or a place.

All branches in git are basically just tags that move (or vice versa), and both are just references with semantics based on their path, either `refs/tags` or `refs/heads` for tags or branches respectively.  So rather than images owning tags, a tag simply points to a tree of images, an album is just a tag, and a person could simply be a tag pointing to a tree of metadata (face locations) that points to images.  You could even trivially add people groups by having a tag that points to a tree of people.

In this schema, the permissions could be at the tag level, as well as the object level.  You can allow someone to be able to see/edit only certain faces, or certain groups of people, or albums, etc.  If so desired, a single image could have permissions as well.  Such a data migration to references and objects wouldn't be trivial, but it would fulfill what I think is the expected functionality of the app in providing multigranular permissions.

I scanned and uploaded 12000 images to Immich before I discovered the permissions limitations, and it's a real shame I can't share this great functionality (privately-hosted facial recognition) with my family.

**Comment by kwongtn** [comment]
+1 on the tag suggestion

would be very useful for, say I want to extract all images that I have of members of a particular group, in which members are of multiple group & categories.

For example if I have a person that is of family, club A and B, I can filter on club A and get everyone's photos there.

**Comment by imagicalFantasy** [comment]
+1 it could be really usefull.
